src/FsAutoComplete.Core/SemaphoreSlimLocks.fs

```diff
@@ -0,0 +1,38 @@
+namespace FsAutoComplete
+
+open System
+open System.Threading.Tasks
+
+/// <summary>
+/// An awaitable wrapper around a task whose result is disposable. The wrapper is not disposable, so this prevents usage errors like "use _lock = myAsync()" when the appropriate usage should be "use! _lock = myAsync())".
+/// </summary>
+[<Struct>]
```

Comment:
You probably don't need comparison and equality here either?

Reply:
Good catch!